db/db_test2.cc

```diff
@@ -5439,6 +5439,104 @@ TEST_F(DBTest2, AutoPrefixMode1) {
     ASSERT_EQ("a1", iterator->key().ToString());
   }
 }
+
+class TestEnvWrapper : public EnvWrapper {
+ public:
+  TestEnvWrapper(Env* _target) : EnvWrapper(_target) {}
+  Status NewWritableFile(const std::string& fname,
+                         std::unique_ptr<WritableFile>* result,
+                         const EnvOptions& env_opts) override {
+    Status s = target()->FileExists(fname);
+    EXPECT_TRUE(s.IsNotFound()) << fname << " already exists.";
```

Comment:
Nit: can this logic live in SpecialEnv and guard by a variable?

Reply:
Sure. Let me update